File: src/main/java-graphql/io/github/graphql/model/GistFileResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for GistFile
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class GistFileResponseProjection extends GraphQLResponseProjection {

    public GistFileResponseProjection() {
    }

    @Override
    public GistFileResponseProjection all$() {
        return all$(3);
    }

    @Override
    public GistFileResponseProjection all$(int maxDepth) {
        this.encodedName();
        this.encoding();
        this.extension();
        this.isImage();
        this.isTruncated();
        if (projectionDepthOnFields.getOrDefault("GistFileResponseProjection.LanguageResponseProjection.language", 0) <= maxDepth) {
            projectionDepthOnFields.put("GistFileResponseProjection.LanguageResponseProjection.language", projectionDepthOnFields.getOrDefault("GistFileResponseProjection.LanguageResponseProjection.language", 0) + 1);
            this.language(new LanguageResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("GistFileResponseProjection.LanguageResponseProjection.language", 0)));
        }
        this.name();
        this.size();
        this.text();
        this.typename();
        return this;
    }

    public GistFileResponseProjection encodedName() {
        return encodedName(null);
    }

    public GistFileResponseProjection encodedName(String alias) {
        fields.add(new GraphQLResponseField("encodedName").alias(alias));
        return this;
    }

    public GistFileResponseProjection encoding() {
        return encoding(null);
    }

    public GistFileResponseProjection encoding(String alias) {
        fields.add(new GraphQLResponseField("encoding").alias(alias));
        return this;
    }

    public GistFileResponseProjection extension() {
        return extension(null);
    }

    public GistFileResponseProjection extension(String alias) {
        fields.add(new GraphQLResponseField("extension").alias(alias));
        return this;
    }

    public GistFileResponseProjection isImage() {
        return isImage(null);
    }

    public GistFileResponseProjection isImage(String alias) {
        fields.add(new GraphQLResponseField("isImage").alias(alias));
        return this;
    }

    public GistFileResponseProjection isTruncated() {
        return isTruncated(null);
    }

    public GistFileResponseProjection isTruncated(String alias) {
        fields.add(new GraphQLResponseField("isTruncated").alias(alias));
        return this;
    }

    public GistFileResponseProjection language(LanguageResponseProjection subProjection) {
        return language(null, subProjection);
    }

    public GistFileResponseProjection language(String alias, LanguageResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("language").alias(alias).projection(subProjection));
        return this;
    }

    public GistFileResponseProjection name() {
        return name(null);
    }

    public GistFileResponseProjection name(String alias) {
        fields.add(new GraphQLResponseField("name").alias(alias));
        return this;
    }

    public GistFileResponseProjection size() {
        return size(null);
    }

    public GistFileResponseProjection size(String alias) {
        fields.add(new GraphQLResponseField("size").alias(alias));
        return this;
    }

    public GistFileResponseProjection text() {
        return text((String)null);
    }

    public GistFileResponseProjection text(String alias) {
        fields.add(new GraphQLResponseField("text").alias(alias));
        return this;
    }

    public GistFileResponseProjection text(GistFileTextParametrizedInput input) {
        return text(null, input);
    }

    public GistFileResponseProjection text(String alias, GistFileTextParametrizedInput input) {
        fields.add(new GraphQLResponseField("text").alias(alias).parameters(input));
        return this;
    }

    public GistFileResponseProjection typename() {
        return typename(null);
    }

    public GistFileResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/graphql/io/github/graphql/model/GistFileResponseProjection.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for GistFile
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-05T19:09:00+0800"
)
open class GistFileResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): GistFileResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): GistFileResponseProjection {
        this.encodedName()
        this.encoding()
        this.extension()
        this.isImage()
        this.isTruncated()
        if (projectionDepthOnFields.getOrDefault("GistFileResponseProjection.LanguageResponseProjection.language", 0) <= maxDepth) {
            projectionDepthOnFields["GistFileResponseProjection.LanguageResponseProjection.language"] = projectionDepthOnFields.getOrDefault("GistFileResponseProjection.LanguageResponseProjection.language", 0) + 1
            this.language(LanguageResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("GistFileResponseProjection.LanguageResponseProjection.language", 0)))
        }
        this.name()
        this.size()
        this.text()
        this.typename()
        return this
    }

    fun encodedName(): GistFileResponseProjection = encodedName(null)

    fun encodedName(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("encodedName").alias(alias))
        return this
    }

    fun encoding(): GistFileResponseProjection = encoding(null)

    fun encoding(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("encoding").alias(alias))
        return this
    }

    fun extension(): GistFileResponseProjection = extension(null)

    fun extension(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("extension").alias(alias))
        return this
    }

    fun isImage(): GistFileResponseProjection = isImage(null)

    fun isImage(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("isImage").alias(alias))
        return this
    }

    fun isTruncated(): GistFileResponseProjection = isTruncated(null)

    fun isTruncated(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("isTruncated").alias(alias))
        return this
    }

    fun language(subProjection: LanguageResponseProjection): GistFileResponseProjection = language(null, subProjection)

    fun language(alias: String?, subProjection: LanguageResponseProjection): GistFileResponseProjection {
        fields.add(GraphQLResponseField("language").alias(alias).projection(subProjection))
        return this
    }

    fun name(): GistFileResponseProjection = name(null)

    fun name(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("name").alias(alias))
        return this
    }

    fun size(): GistFileResponseProjection = size(null)

    fun size(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("size").alias(alias))
        return this
    }

    fun text(): GistFileResponseProjection = text(null)

    fun text(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("text").alias(alias))
        return this
    }

    fun text(input: GistFileTextParametrizedInput): GistFileResponseProjection = text(null, input)

    fun text(alias: String?, input: GistFileTextParametrizedInput): GistFileResponseProjection {
        fields.add(GraphQLResponseField("text").alias(alias).parameters(input))
        return this
    }

    fun typename(): GistFileResponseProjection = typename(null)

    fun typename(alias: String?): GistFileResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }


}
